### ISSUE TITLE
OpenSSL 0.9.8 - Homebrew

### ISSUE THREAD

**Issue by DomT4** [rvm/rvm#3618]
We removed `openssl098` from Homebrew/Versions as per https://github.com/Homebrew/homebrew-versions/commit/4e169f22cc48e61e2ed6e2f80a5414fe281db335. We've been warning that we would do so for 6 months or so now, and OpenSSL themselves have been warning 0.9.8 was coming up to EOL for longer than that, so hopefully this isn't too much of a surprise.

Is there a specific reason RVM builds the 1.8.x series [against](https://github.com/rvm/rvm/blob/master/scripts/functions/requirements/osx_brew#L176-L183) an OpenSSL that old? It was added in https://github.com/rvm/rvm/commit/7230b9aed0e6c5e05cb368ef0b4a2910de161e43.

Over at Homebrew we have a Ruby 1.8.7 formula that currently builds against the OpenSSL 1.0.2 branch without the need for any [patching](https://github.com/Homebrew/homebrew-versions/blob/master/ruby187.rb) to make it do so.

We also support the OpenSSL 1.0.1 branch (Until likely January 2017, in line with the OpenSSL support deprecation schedule) if greater insurance against potentially breaking upstream changes is what was intended in the building against the 098 formula.

Happy to submit a PR to build against a modern OpenSSL if you'd like one, but wanted to try and establish the chain of why the setup is as it is in case we're missing an edge case, etc.

**Comment by rys** [comment]
I've created PR 3619 to remove the OpenSSL 0.9.8 dependency from our OS X builds. I think it's fine, so I'll likely merge after a bit more testing, but I don't have complete visibility of what it could possibly break. We've got no good tests for it, but neither am I too bothered if it does break because we shouldn't be building against broken and old OpenSSL releases anyway.

https://github.com/rvm/rvm/pull/3619
